homeassistant/components/mobile_app/webhook.py

```diff
@@ -446,6 +446,26 @@ async def webhook_register_sensor(hass, config_entry, data):
             "Re-register for %s of existing sensor %s", device_name, unique_id
         )
 
+        entry = entity_registry.async_get(existing_sensor)
+        changes = {}
+
+        if (
+            new_name := f"{device_name} {data[ATTR_SENSOR_NAME]}"
+        ) != entry.original_name:
+            changes["original_name"] = new_name
+
+        for ent_reg_key, data_key in (
+            ("device_class", ATTR_SENSOR_DEVICE_CLASS),
+            ("unit_of_measurement", ATTR_SENSOR_UOM),
+            ("entity_category", ATTR_SENSOR_ENTITY_CATEGORY),
+            ("original_icon", ATTR_SENSOR_ICON),
+        ):
+            if data_key in data and getattr(entry, ent_reg_key) != data[data_key]:
+                changes[ent_reg_key] = data[data_key]
+
+        if changes:
+            er.async_get(hass).async_update_entity(existing_sensor, **changes)
```

Comment:
```suggestion
            entity_registry.async_update_entity(existing_sensor, **changes)
```

Looks like we already have it from above?

Maybe update that one to not await

Reply:
I already did that in https://github.com/home-assistant/core/pull/58542